agent/containers/images/visualizers/live-metric-visualizer/prom_ds.py

```diff
@@ -0,0 +1,92 @@
+import json
+import requests
+import subprocess
+import time
+
+graf_base = "http://localhost:3000/"
+
+while True:
+    try:
+        response = requests.get(graf_base)
+        if response.status_code == 200:
+            break
+    except Exception:
+        time.sleep(0.1)
+
+tokenholder = open("key.txt", "w")
+args = [
+    "curl",
+    "http://localhost:3000/api/auth/keys",
+    "-XPOST",
+    "-uadmin:admin",
+    "-H",
+    "Content-Type: application/json",
+    "-d",
+    '{"role":"Admin","name":"new_api_key"}',
+]
+process = subprocess.run(args, stdout=tokenholder)
+tokenholder.close()
+
+tokenholder = open("key.txt", "r")
+token_raw = tokenholder.readline()
+tokenholder.close()
+token_dict = json.loads(token_raw)
+print(token_dict["key"])
+token = token_dict["key"]
```

Comment:
Doesn't seem there's a need for `process` since you don't use it again.

I'm a little curious, too, why you resorted to `curl` and all that work to JSONify stdout, when you used `requests` below??